File: frontend/virtual-gallery-client/src/types/exhibition.ts
export interface Exhibition {
  id: string;
  title: string;
  theme: string;
  description: string;
  submissionStartDate: string;
  submissionEndDate: string;
  startDate: string;
  endDate: string;
  createdAt: string;
  status: string;
  artworkCount: number;
}

export interface ExhibitionCalendarItem {
  id: string;
  title: string;
  theme: string;
  submissionStartDate: string;
  submissionEndDate: string;
  startDate: string;
  endDate: string;
  status: string;
}

export interface ExhibitionArtworkItem {
  id: string;
  title: string;
  description: string;
  year: number;
  imageUrl: string;
  authorName: string;
  authorEmail: string;
  tags: string[];
}

export interface ExhibitionDetails {
  id: string;
  title: string;
  theme: string;
  description: string;
  submissionStartDate: string;
  submissionEndDate: string;
  startDate: string;
  endDate: string;
  createdAt: string;
  status: string;
  artworkCount: number;
  artworks: ExhibitionArtworkItem[];
}

File: backend/VirtualGallery.Api/Models/Exhibition.cs
using System.ComponentModel.DataAnnotations;

namespace VirtualGallery.Api.Models;

public class Exhibition
{
    public Guid Id { get; set; } = Guid.NewGuid();

    [Required]
    [MaxLength(200)]
    public string Title { get; set; } = string.Empty;

    [Required]
    [MaxLength(200)]
    public string Theme { get; set; } = string.Empty;

    [Required]
    [MaxLength(2000)]
    public string Description { get; set; } = string.Empty;

    public DateTime SubmissionStartDate { get; set; }
    public DateTime SubmissionEndDate { get; set; }
    public DateTime StartDate { get; set; }
    public DateTime EndDate { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public ICollection<Submission> Submissions { get; set; } = new List<Submission>();
    public ICollection<Artwork> Artworks { get; set; } = new List<Artwork>();
}